Source/player.cpp

```diff
@@ -798,6 +789,18 @@ bool DoAttack(Player &player)
 				didhit = PlrHitObj(player, *object);
 			}
 		}
+
+		if (!gbIsHellfire || !HasAllOf(player._pIFlags, ItemSpecialEffect::FireDamage | ItemSpecialEffect::LightningDamage)) {
+			if (HasAnyOf(player._pIFlags, ItemSpecialEffect::FireDamage)) {
+				int elementalDamage = didhit ? RandomIntBetween(player._pIFMinDam, player._pIFMaxDam) : 0;
+				AddMissile(position, { 1, 0 }, Direction::South, MissileID::WeaponExplosion, TARGET_MONSTERS, player, elementalDamage, 0);

```

Comment:
`didhit` is calculated by `PlrHitMonst` / `PlrHitPlr` and this calculates if the hit can fail. This check is used to calculate the damage (0 if the hit failed).
In `AddWeaponExplosion` we call `CheckMissileCol` and this calls `Plr2PlrMHit` / `MonsterMHit`. These functions also calculate if the missile hits or not.
Does the weapon explosion now have to pass two hit checks instead of one? 🤔

Reply:
Yes, I did this on purpose to match the behavior of elemental arrows. For arrows, the elemental explosion can't even spawn unless the arrow connects with a target so we can't really fix that behavior without refactoring the collision logic or adding yet another parameter.

Reply:
I looked at the arrow logic and I'm with you.
It just showed me that we are changing / nerfing the elemental damage. But I'm fine with aligning it with the way elemental arrows are handled.